# ICC Bowler statistics

###### Objectives:

1.Extracting new information from a column,
2.Create a column based on a condition or function,
3.Removing a string from a column,
4.Checking the unique values for each column,
5.performing calculation in dataframe columns,
6.dataframe sorting,
7.dataframe slicing.

Questions: 
1. Remove the columns BBI and BBM.
2. How many players played for ICC?
3. How many different countries are present in this dataset? 
4. Which player(s) had played for the longest period of time?
5. Which player(s) had played for the shortest period of time?
6. How many Australian Bowlers are present in this dataset?
7. Is there any Bangladeshi player present in this dataset?
8. Which player had the lowest economy rate?
9. Which player had the lowest strike rate?
10. Which player had the lowest bowling average?

###### References:
[1] Data Source: https://stats.espncricinfo.com/ci/content/records/223646.html

[2] Dataset : https://github.com/SKawsar/Data_Analysis_with_Python

In [1186]:
import pandas as pd
import numpy as np

In [1187]:
df =pd.read_excel("test_cricket.xlsx",sheet_name='wickets')
display(df.head())

Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0  M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180   800   
1           SK Warne (AUS)  1992-2007  145   273  40705  17995   708   
2         A Kumble (INDIA)  1990-2008  132   236  40850  18355   619   
3        JM Anderson (ENG)  2003-2021  162   301  34791  16457   617   
4         GD McGrath (AUS)  1993-2007  124   243  29248  12186   563   

                   BBI                  BBM    Ave  Econ    SR   5  10  
0  1951-09-01 00:00:00               16/220  22.72  2.47  55.0  67  22  
1  1971-08-01 00:00:00               12/128  25.41  2.65  57.4  37  10  
2  1974-10-01 00:00:00               14/149  29.65  2.69  65.9  35   8  
3  1942-07-01 00:00:00  1971-11-01 00:00:00  26.67  2.83  56.3  30   3  
4  2021-08-24 00:00:00  2021-10-27 00:00:00  21.64  2.49  51.9  29   3

## Remove the column BBI and BBM

In [1188]:
df=df.drop(columns=['BBI','BBM'],axis=1)
display(df.head())

Player       Span  Mat  Inns  Balls   Runs  Wkts    Ave  \
0  M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180   800  22.72   
1           SK Warne (AUS)  1992-2007  145   273  40705  17995   708  25.41   
2         A Kumble (INDIA)  1990-2008  132   236  40850  18355   619  29.65   
3        JM Anderson (ENG)  2003-2021  162   301  34791  16457   617  26.67   
4         GD McGrath (AUS)  1993-2007  124   243  29248  12186   563  21.64   

   Econ    SR   5  10  
0  2.47  55.0  67  22  
1  2.65  57.4  37  10  
2  2.69  65.9  35   8  
3  2.83  56.3  30   3  
4  2.49  51.9  29   3

## Number of players played for ICC

In [1189]:
file = df.copy()
file['Player'] = file['Player'].str.replace("(", "")
file['Player'] = file['Player'].str.replace(")", "")
file['Player'] = file['Player'].str.replace("\xa0", " ")

display(file.head())

Player       Span  Mat  Inns  Balls   Runs  Wkts    Ave  \
0  M Muralitharan ICC/SL  1992-2010  133   230  44039  18180   800  22.72   
1           SK Warne AUS  1992-2007  145   273  40705  17995   708  25.41   
2         A Kumble INDIA  1990-2008  132   236  40850  18355   619  29.65   
3        JM Anderson ENG  2003-2021  162   301  34791  16457   617  26.67   
4         GD McGrath AUS  1993-2007  124   243  29248  12186   563  21.64   

   Econ    SR   5  10  
0  2.47  55.0  67  22  
1  2.65  57.4  37  10  
2  2.69  65.9  35   8  
3  2.83  56.3  30   3  
4  2.49  51.9  29   3

In [1190]:
file_player = file['Player'].str.split(" ")
file_player = pd.DataFrame(file_player)

file_player['Country'] = file_player['Player'].str[-1]

print(type(file_player))

display(file_player.head())

<class 'pandas.core.frame.DataFrame'>


Player Country
0  [M, Muralitharan, ICC/SL]  ICC/SL
1           [SK, Warne, AUS]     AUS
2         [A, Kumble, INDIA]   INDIA
3        [JM, Anderson, ENG]     ENG
4         [GD, McGrath, AUS]     AUS

In [1191]:
df = pd.concat([df, file_player], axis=1)
display(df.head())

Player       Span  Mat  Inns  Balls   Runs  Wkts    Ave  \
0  M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180   800  22.72   
1           SK Warne (AUS)  1992-2007  145   273  40705  17995   708  25.41   
2         A Kumble (INDIA)  1990-2008  132   236  40850  18355   619  29.65   
3        JM Anderson (ENG)  2003-2021  162   301  34791  16457   617  26.67   
4         GD McGrath (AUS)  1993-2007  124   243  29248  12186   563  21.64   

   Econ    SR   5  10                     Player Country  
0  2.47  55.0  67  22  [M, Muralitharan, ICC/SL]  ICC/SL  
1  2.65  57.4  37  10           [SK, Warne, AUS]     AUS  
2  2.69  65.9  35   8         [A, Kumble, INDIA]   INDIA  
3  2.83  56.3  30   3        [JM, Anderson, ENG]     ENG  
4  2.49  51.9  29   3         [GD, McGrath, AUS]     AUS

In [1192]:
def icc_check(x):
    if "ICC" in x:
        return "Yes"
    else:
        return "No"
df['played_for_ICC'] = df['Country'].apply(icc_check)
display(df.head(10))

Player       Span  Mat  Inns  Balls   Runs  Wkts    Ave  \
0  M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180   800  22.72   
1           SK Warne (AUS)  1992-2007  145   273  40705  17995   708  25.41   
2         A Kumble (INDIA)  1990-2008  132   236  40850  18355   619  29.65   
3        JM Anderson (ENG)  2003-2021  162   301  34791  16457   617  26.67   
4         GD McGrath (AUS)  1993-2007  124   243  29248  12186   563  21.64   
5          SCJ Broad (ENG)  2007-2021  148   272  29713  14502   523  27.72   
6            CA Walsh (WI)  1984-2001  132   242  30019  12688   519  24.44   
7            DW Steyn (SA)  2004-2019   93   171  18608  10077   439  22.95   
8      N Kapil Dev (INDIA)  1978-1994  131   227  27740  12867   434  29.64   
9        HMRKB Herath (SL)  1999-2018   93   170  25993  12157   433  28.07   

   Econ    SR   5  10                     Player Country played_for_ICC  
0  2.47  55.0  67  22  [M, Muralitharan, ICC/SL]  ICC/SL            Yes  
1  2.65  57.4  37  10           [SK, Warne, AUS]     AUS             No  
2  2.69  65.9  35   8         [A, Kumble, INDIA]   INDIA             No  
3  2.83  56.3  30   3        [JM, Anderson, ENG]     ENG             No  
4  2.49  51.9  29   3         [GD, McGrath, AUS]     AUS             No  
5  2.92  56.8  18   3          [SCJ, Broad, ENG]     ENG             No  
6  2.53  57.8  22   3            [CA, Walsh, WI]      WI             No  
7  3.24  42.3  26   5            [DW, Steyn, SA]      SA             No  
8  2.78  63.9  23   2     [N, Kapil, Dev, INDIA]   INDIA             No  
9  2.80  60.0  34   9        [HMRKB, Herath, SL]      SL             No

In [1193]:
df['played_for_ICC'].value_counts()

No     74
Yes     5
Name: played_for_ICC, dtype: int64

###### Number of players plyed for ICC = 5

## Number of Australian Bowlers and presence of  Bangladeshi player  in the dataset

In [1194]:
df['Country'] = df['Country'].str.replace("ICC/", "")
display(df.head())
df['Country'].value_counts()

Player       Span  Mat  Inns  Balls   Runs  Wkts    Ave  \
0  M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180   800  22.72   
1           SK Warne (AUS)  1992-2007  145   273  40705  17995   708  25.41   
2         A Kumble (INDIA)  1990-2008  132   236  40850  18355   619  29.65   
3        JM Anderson (ENG)  2003-2021  162   301  34791  16457   617  26.67   
4         GD McGrath (AUS)  1993-2007  124   243  29248  12186   563  21.64   

   Econ    SR   5  10                     Player Country played_for_ICC  
0  2.47  55.0  67  22  [M, Muralitharan, ICC/SL]      SL            Yes  
1  2.65  57.4  37  10           [SK, Warne, AUS]     AUS             No  
2  2.69  65.9  35   8         [A, Kumble, INDIA]   INDIA             No  
3  2.83  56.3  30   3        [JM, Anderson, ENG]     ENG             No  
4  2.49  51.9  29   3         [GD, McGrath, AUS]     AUS             No

AUS        18
ENG        13
INDIA      10
WI          9
SA          8
NZ          7
PAK         7
SL          3
ENG/ICC     2
BDESH       1
ZIM         1
Name: Country, dtype: int64

###### Number of Australian bowler 18,
Yes there is presence of Bangladeshi player.

## Player(s) playing for the longest and  shortest period of time

In [1195]:
# splitting the 'Span' column based on the "-"
df_span = df['Span'].str.split("-", expand=True)

# conccatinating the new dataframe with the main dataframe
df = pd.concat([df, df_span], axis=1)

# renaming the newly created column names
df = df.rename(columns={0: "start_year",
                        1: "end_year"})

# removing the "Span" column
df = df.drop("Span", axis=1)

display(df.head())

Player  Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR  \
0  M Muralitharan (ICC/SL)  133   230  44039  18180   800  22.72  2.47  55.0   
1           SK Warne (AUS)  145   273  40705  17995   708  25.41  2.65  57.4   
2         A Kumble (INDIA)  132   236  40850  18355   619  29.65  2.69  65.9   
3        JM Anderson (ENG)  162   301  34791  16457   617  26.67  2.83  56.3   
4         GD McGrath (AUS)  124   243  29248  12186   563  21.64  2.49  51.9   

    5  10                     Player Country played_for_ICC start_year  \
0  67  22  [M, Muralitharan, ICC/SL]      SL            Yes       1992   
1  37  10           [SK, Warne, AUS]     AUS             No       1992   
2  35   8         [A, Kumble, INDIA]   INDIA             No       1990   
3  30   3        [JM, Anderson, ENG]     ENG             No       2003   
4  29   3         [GD, McGrath, AUS]     AUS             No       1993   

  end_year  
0     2010  
1     2007  
2     2008  
3     2021  
4     2007

In [1196]:
df['start_year'] = df['start_year'].astype('int') 
df['end_year'] = df['end_year'].astype('int')

print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Player          79 non-null     object 
 1   Mat             79 non-null     int64  
 2   Inns            79 non-null     int64  
 3   Balls           79 non-null     int64  
 4   Runs            79 non-null     int64  
 5   Wkts            79 non-null     int64  
 6   Ave             79 non-null     float64
 7   Econ            79 non-null     float64
 8   SR              79 non-null     float64
 9   5               79 non-null     int64  
 10  10              79 non-null     int64  
 11  Player          79 non-null     object 
 12  Country         79 non-null     object 
 13  played_for_ICC  79 non-null     object 
 14  start_year      79 non-null     int32  
 15  end_year        79 non-null     int32  
dtypes: float64(3), int32(2), int64(7), object(4)
memory usage: 9.4+ KB
None


In [1197]:
df['years_played'] = df['end_year'] - df['start_year']
df = df.drop(['start_year', "end_year"], axis=1)
display(df.head(10))

Player  Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR  \
0  M Muralitharan (ICC/SL)  133   230  44039  18180   800  22.72  2.47  55.0   
1           SK Warne (AUS)  145   273  40705  17995   708  25.41  2.65  57.4   
2         A Kumble (INDIA)  132   236  40850  18355   619  29.65  2.69  65.9   
3        JM Anderson (ENG)  162   301  34791  16457   617  26.67  2.83  56.3   
4         GD McGrath (AUS)  124   243  29248  12186   563  21.64  2.49  51.9   
5          SCJ Broad (ENG)  148   272  29713  14502   523  27.72  2.92  56.8   
6            CA Walsh (WI)  132   242  30019  12688   519  24.44  2.53  57.8   
7            DW Steyn (SA)   93   171  18608  10077   439  22.95  3.24  42.3   
8      N Kapil Dev (INDIA)  131   227  27740  12867   434  29.64  2.78  63.9   
9        HMRKB Herath (SL)   93   170  25993  12157   433  28.07  2.80  60.0   

    5  10                     Player Country played_for_ICC  years_played  
0  67  22  [M, Muralitharan, ICC/SL]      SL            Yes            18  
1  37  10           [SK, Warne, AUS]     AUS             No            15  
2  35   8         [A, Kumble, INDIA]   INDIA             No            18  
3  30   3        [JM, Anderson, ENG]     ENG             No            18  
4  29   3         [GD, McGrath, AUS]     AUS             No            14  
5  18   3          [SCJ, Broad, ENG]     ENG             No            14  
6  22   3            [CA, Walsh, WI]      WI             No            17  
7  26   5            [DW, Steyn, SA]      SA             No            15  
8  23   2     [N, Kapil, Dev, INDIA]   INDIA             No            16  
9  34   9        [HMRKB, Herath, SL]      SL             No            19

In [1198]:
df.sort_values(by='years_played', ascending = False).head(10)

Player  Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR  \
21         Imran Khan (PAK)   88   142  19458   8258   362  22.81  2.54  53.7   
55           GS Sobers (WI)   93   159  21599   7999   235  34.03  2.22  91.9   
9         HMRKB Herath (SL)   93   170  25993  12157   433  28.07  2.80  60.0   
0   M Muralitharan (ICC/SL)  133   230  44039  18180   800  22.72  2.47  55.0   
2          A Kumble (INDIA)  132   236  40850  18355   619  29.65  2.69  65.9   
3         JM Anderson (ENG)  162   301  34791  16457   617  26.67  2.83  56.3   
37       JH Kallis (ICC/SA)  166   272  20232   9535   292  32.65  2.82  69.2   
32            LR Gibbs (WI)   79   148  27115   8989   309  29.09  1.98  87.7   
13        Wasim Akram (PAK)  104   181  22627   9779   414  23.62  2.59  54.6   
22      DL Vettori (ICC/NZ)  113   187  28814  12441   362  34.36  2.59  79.5   

     5  10                     Player Country played_for_ICC  years_played  
21  23   6         [Imran, Khan, PAK]     PAK             No            21  
55   6   0           [GS, Sobers, WI]      WI             No            20  
9   34   9        [HMRKB, Herath, SL]      SL             No            19  
0   67  22  [M, Muralitharan, ICC/SL]      SL            Yes            18  
2   35   8         [A, Kumble, INDIA]   INDIA             No            18  
3   30   3        [JM, Anderson, ENG]     ENG             No            18  
37   5   0       [JH, Kallis, ICC/SA]      SA            Yes            18  
32  18   2            [LR, Gibbs, WI]      WI             No            18  
13  25   5        [Wasim, Akram, PAK]     PAK             No            17  
22  20   3      [DL, Vettori, ICC/NZ]      NZ            Yes            17

###### Player playin for longest period of time : Imran khan (20 Years)

In [1199]:
df.sort_values(by='years_played', ascending = True).head(10)

Player  Mat  Inns  Balls  Runs  Wkts    Ave  Econ    SR  \
44         GP Swann (ENG)   60   109  15349  7642   255  29.96  2.98  60.1   
71          K Rabada (SA)   47    86   8785  4846   213  22.75  3.30  41.2   
54       Yasir Shah (PAK)   45    83  13601  7247   235  30.83  3.19  57.8   
61  SJ Harmison (ENG/ICC)   63   115  13375  7192   226  31.82  3.22  59.1   
72     JR Hazlewood (AUS)   55   103  11887  5438   212  25.65  2.74  56.0   
28       MG Johnson (AUS)   73   140  16001  8891   313  28.40  3.33  51.1   
48       MJ Hoggard (ENG)   67   122  13909  7564   248  30.50  3.26  56.0   
59          D Gough (ENG)   58    95  11821  6503   229  28.39  3.30  51.6   
51        AV Bedser (ENG)   51    92  15918  5876   236  24.89  2.21  67.4   
63      VD Philander (SA)   64   119  11391  5000   224  22.32  2.63  50.8   

     5  10                   Player  Country played_for_ICC  years_played  
44  17   3         [GP, Swann, ENG]      ENG             No             5  
71  10   4          [K, Rabada, SA]       SA             No             6  
54  16   3       [Yasir, Shah, PAK]      PAK             No             7  
61   8   1  [SJ, Harmison, ENG/ICC]  ENG/ICC            Yes             7  
72   9   0     [JR, Hazlewood, AUS]      AUS             No             7  
28  12   3       [MG, Johnson, AUS]      AUS             No             8  
48   7   1       [MJ, Hoggard, ENG]      ENG             No             8  
59   9   0          [D, Gough, ENG]      ENG             No             9  
51  15   5        [AV, Bedser, ENG]      ENG             No             9  
63  13   2      [VD, Philander, SA]       SA             No             9

###### Player playin for shortest period of time : GP Swann ( 5 Years)

## Player having the lowest economy rate

In [1200]:
df.sort_values(by='Econ', ascending = True).head(10)

Player  Mat  Inns  Balls  Runs  Wkts    Ave  Econ    SR   5  \
32       LR Gibbs (WI)   79   148  27115  8989   309  29.09  1.98  87.7  18   
47      R Benaud (AUS)   63   116  19108  6704   248  27.03  2.10  77.0  16   
35  DL Underwood (ENG)   86   151  21862  7674   297  25.83  2.10  73.6  17   
39     BS Bedi (INDIA)   67   118  21364  7637   266  28.71  2.14  80.3  14   
68   CV Grimmett (AUS)   37    67  14513  5231   216  24.21  2.16  67.1  21   
51     AV Bedser (ENG)   51    92  15918  5876   236  24.89  2.21  67.4  15   
55      GS Sobers (WI)   93   159  21599  7999   235  34.03  2.22  91.9   6   
60   RR Lindwall (AUS)   61   113  13650  5251   228  23.03  2.30  59.8  12   
15    CEL Ambrose (WI)   98   179  22103  8501   405  20.99  2.30  54.5  22   
45    JB Statham (ENG)   70   129  16056  6261   252  24.84  2.33  63.7   9   

    10                Player Country played_for_ICC  years_played  
32   2       [LR, Gibbs, WI]      WI             No            18  
47   1      [R, Benaud, AUS]     AUS             No            12  
35   6  [DL, Underwood, ENG]     ENG             No            16  
39   1     [BS, Bedi, INDIA]   INDIA             No            13  
68   7   [CV, Grimmett, AUS]     AUS             No            11  
51   5     [AV, Bedser, ENG]     ENG             No             9  
55   0      [GS, Sobers, WI]      WI             No            20  
60   0   [RR, Lindwall, AUS]     AUS             No            14  
15   3    [CEL, Ambrose, WI]      WI             No            12  
45   1    [JB, Statham, ENG]     ENG             No            14

###### Player having the lowest economy rate : LR Gibbs (1.98)

## Player having the lowest strike rate

In [1201]:
df.sort_values(by='SR', ascending = True).head(10)

Player  Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR   5  \
71       K Rabada (SA)   47    86   8785   4846   213  22.75  3.30  41.2  10   
7        DW Steyn (SA)   93   171  18608  10077   439  22.95  3.24  42.3  26   
20  Waqar Younis (PAK)   87   154  16224   8788   373  23.56  3.25  43.4  22   
19    MD Marshall (WI)   81   151  17584   7876   376  20.94  2.68  46.7  22   
25      AA Donald (SA)   72   129  15519   7344   330  22.25  2.83  47.0  20   
43      MA Starc (AUS)   61   117  12575   7031   255  27.57  3.35  49.3  13   
33    FS Trueman (ENG)   67   127  15178   6625   307  21.57  2.61  49.4  17   
63   VD Philander (SA)   64   119  11391   5000   224  22.32  2.63  50.8  13   
10  Sir RJ Hadlee (NZ)   86   150  21918   9611   431  22.29  2.63  50.8  36   
41       J Garner (WI)   58   111  13169   5433   259  20.97  2.47  50.8   7   

    10                 Player Country played_for_ICC  years_played  
71   4        [K, Rabada, SA]      SA             No             6  
7    5        [DW, Steyn, SA]      SA             No            15  
20   5   [Waqar, Younis, PAK]     PAK             No            14  
19   4     [MD, Marshall, WI]      WI             No            13  
25   3       [AA, Donald, SA]      SA             No            10  
43   2       [MA, Starc, AUS]     AUS             No            10  
33   3     [FS, Trueman, ENG]     ENG             No            13  
63   2    [VD, Philander, SA]      SA             No             9  
10   9  [Sir, RJ, Hadlee, NZ]      NZ             No            17  
41   0        [J, Garner, WI]      WI             No            10

###### Player having the lowest strike rate : K Rabada (41.2)

## Player having the lowest bowling average

In [1202]:
df.sort_values(by='Ave', ascending = True).head(10)

Player  Mat  Inns  Balls   Runs  Wkts    Ave  Econ    SR  \
19         MD Marshall (WI)   81   151  17584   7876   376  20.94  2.68  46.7   
41            J Garner (WI)   58   111  13169   5433   259  20.97  2.47  50.8   
15         CEL Ambrose (WI)   98   179  22103   8501   405  20.99  2.30  54.5   
33         FS Trueman (ENG)   67   127  15178   6625   307  21.57  2.61  49.4   
4          GD McGrath (AUS)  124   243  29248  12186   563  21.64  2.49  51.9   
25           AA Donald (SA)   72   129  15519   7344   330  22.25  2.83  47.0   
10       Sir RJ Hadlee (NZ)   86   150  21918   9611   431  22.29  2.63  50.8   
63        VD Philander (SA)   64   119  11391   5000   224  22.32  2.63  50.8   
0   M Muralitharan (ICC/SL)  133   230  44039  18180   800  22.72  2.47  55.0   
71            K Rabada (SA)   47    86   8785   4846   213  22.75  3.30  41.2   

     5  10                     Player Country played_for_ICC  years_played  
19  22   4         [MD, Marshall, WI]      WI             No            13  
41   7   0            [J, Garner, WI]      WI             No            10  
15  22   3         [CEL, Ambrose, WI]      WI             No            12  
33  17   3         [FS, Trueman, ENG]     ENG             No            13  
4   29   3         [GD, McGrath, AUS]     AUS             No            14  
25  20   3           [AA, Donald, SA]      SA             No            10  
10  36   9      [Sir, RJ, Hadlee, NZ]      NZ             No            17  
63  13   2        [VD, Philander, SA]      SA             No             9  
0   67  22  [M, Muralitharan, ICC/SL]      SL            Yes            18  
71  10   4            [K, Rabada, SA]      SA             No             6

###### Player having the lowest bowling average: MD Marshall (20.94)